spark/src/main/scala/org/apache/spark/delta/sharing/PreSignedUrlCache.scala

```diff
@@ -81,9 +95,21 @@ class CachedTableManager(
         logInfo(s"Updating pre signed urls for $tablePath (expiration time: " +
           s"${new java.util.Date(cachedTable.expiration)})")
         try {
+          val (idToUrl, expiration) = cachedTable.refresher()
+          // scalastyle:off println
+          Console.println(s"----[linzhou]------refreshed, timeToExpier:" +
+            s"${expiration - System.currentTimeMillis()}")
           val newTable = new CachedTable(
-            preSignedUrlExpirationMs + System.currentTimeMillis(),
-            cachedTable.refresher(),
+            if (isValidUrlExpirationTime(expiration)) {
+              Console.println(s"----[linzhou]------valid:" +
+                s"${expiration - System.currentTimeMillis()}")
+              expiration
```

Comment:
A question is, if the customer by anychance set `preSignedUrlExpirationMs` to a smaller number, do we respect `preSignedUrlExpirationMs` or still respect `expiration` of the urls

@chakankardb @zsxwing

Reply:
Can we pick the more conservative value between the two?

Reply:
no, a conservative value (if you mean larger value) may result in an expired url.

Reply:
@linzhou-db In this context, conservative would be min... :)

Reply:
sg, updated.